### PR TITLE
Backend user settings

### PR DESCRIPTION
**Pull request by jniemii**

### CODE REVIEW

File: backend/src/db/queries.ts [hunk 1]
```diff
@@ -53,12 +53,15 @@ export const selectProjectData = async (
...
+): Promise<object | null> => {
     const text = 'SELECT settings FROM users WHERE id = $1';
     const values = [userID];
-    const res = await executeQuery(text, values);
-    return res[0] as { settings: object };
-    //return res[0];
+    const res = (await executeQuery(text, values)) as {
+        settings: object | null;
+    }[];
+    const settings = res[0].settings;
```
Comment (itulkki) [inline backend/src/db/queries.ts:68]: Now this function throws an execption if the user does not exist. Please document this behaviour in comments above the function.

File: backend/src/routers/user.ts [hunk 2]
```diff
@@ -93,4 +95,49 @@ userRouter.get(
...
+userRouter.get(
+    '/settings/',
+    expressAsyncHandler(async (req, res) => {
+        const payload = await readToken(req);
+        if (payload === null) {
+            res.status(401).send('No valid token on the request found');
+            return;
+        }
+
+        const data = await selectUserSettings(payload.userID);
```
Comment (itulkki) [inline backend/src/routers/user.ts:107]: Add error handling for this function call or change the behaviour of the function.

Comment (jniemii) [reply backend/src/routers/user.ts:107]: readToken already verifies the user is valid, the only way selectUserSettings would throw error if database connection is broken or the user is deleted and theres a lingering invalid jwt token.

**Review by itulkki — APPROVED** [review]
I will probably do some refactoring to the tests once I got my own pull request working and accepted but those are good for now.

**Pull request merged**